charts/pulsar/templates/zookeeper/zookeeper-cluster.yaml

```diff
@@ -0,0 +1,223 @@
+                  - key: "release"
+                    operator: In
+                    values:
+                      - {{ .Release.Name }}
+                  - key: "component"
+                    operator: In
+                    values:
+                      - {{ .Values.zookeeper.component }}
+              topologyKey: "kubernetes.io/hostname"
+          {{ end }}
+        {{- end }}
+    {{- if .Values.zookeeper.resources }}
+    resources:
+      requests:
+        memory: "{{ .Values.zookeeper.resources.requests.memory }}"
+        cpu: "{{ .Values.zookeeper.resources.requests.cpu }}"
+    {{- end }}
+    {{- if .Values.zookeeper.securityContext }}
+    securityContext: {{ .Values.zookeeper.securityContext }}
+    {{- end }}
+    {{- if .Values.zookeeper.tolerations }}
+    tolerations:
+{{ toYaml .Values.zookeeper.tolerations | indent 6 }}
+    {{- end }}
+    terminationGracePeriodSeconds: {{ .Values.zookeeper.gracePeriod }}
+    jvmOptions:
+      memoryOptions:
+        - {{ .Values.zookeeper.configData.PULSAR_MEM }}
+      gcOptions:
+        - {{ .Values.zookeeper.configData.PULSAR_GC }}
+  {{- if and .Values.volumes.persistence .Values.zookeeper.volumes.persistence }}
+  persistence:
+    data:
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: {{ .Values.zookeeper.volumes.data.size }}
+    {{- if and .Values.volumes.local_storage .Values.zookeeper.volumes.data.local_storage }}
+      storageClassName: "local-storage"
+    {{- else }}
+      {{- if  .Values.zookeeper.volumes.data.storageClass }}
+      storageClassName: "{{ template "pulsar.fullname" . }}-{{ .Values.zookeeper.component }}-{{ .Values.zookeeper.volumes.data.name }}"
+      {{- else if .Values.zookeeper.volumes.data.storageClassName }}
+      storageClassName: {{ .Values.zookeeper.volumes.data.storageClassName }}
+      {{- end -}}
+    {{- end }}
+    dataLog:
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          {{- if .Values.zookeeper.volumes.useSeparateDiskForTxlog }}
+          storage: {{ .Values.zookeeper.volumes.dataLog.size }}
+          {{- else }}
+          storage: "0"
+          {{- end }}
+    {{- if and .Values.volumes.local_storage .Values.zookeeper.volumes.data.local_storage }}
+      storageClassName: "local-storage"
+    {{- else }}
+      {{- if  .Values.zookeeper.volumes.data.storageClass }}
+      storageClassName: "{{ template "pulsar.fullname" . }}-{{ .Values.zookeeper.component }}-{{ .Values.zookeeper.volumes.data.name }}"
+      {{- else if .Values.zookeeper.volumes.data.storageClassName }}
+      storageClassName: {{ .Values.zookeeper.volumes.data.storageClassName }}
+      {{- end -}}
+    {{- end }}
+    reclaimPolicy: Delete
+  {{- end }}
+  config:
+    serverCnxnFactory: org.apache.zookeeper.server.NettyServerCnxnFactory
+    # copy from configmap
+    custom:
+      dataDir: /pulsar/data/zookeeper
```

Comment:
Maybe `dataDir` & `dataLogDir` should be configured only when `.Values.zookeeper.operator.adopt_existing` is true?

Reply:
I remember adding the check as you mentioned that to me before, somehow the change was lost, I'll add it here.

Reply:
oh nvm, I only add for bk